File: 501 - Find Mode in Binary Search Tree.java
import java.util.ArrayList;

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    ArrayList<Integer> ans = new ArrayList<>();
    int curnum, count, maxcount;
    public int[] findMode(TreeNode root) {
        dfs(root);
        int[] modes = new int[ans.size()];
        for (int i = 0; i < ans.size(); i++) {
            modes[i] = ans.get(i);
        }
        return modes;
    }
    public void dfs(TreeNode root) {
        if (root != null){
            dfs(root.left);
            update(root);
            dfs(root.right);
        }
    }
    public void update(TreeNode node) {
        if (node.val == curnum) {
            count++;
        } else {
            curnum = node.val;
            count = 1;
        }
        if (count == maxcount) {
            ans.add(node.val);
        } else if (count > maxcount) {
            maxcount = count;
            ans.clear();
            ans.add(node.val);
        }
    }
}

File: 501 - Find Mode in Binary Search Tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var findMode = function(root) {
    let [count , maxcount, curnum] = [0,0,0];
    let ans = [];
    function dfs(o) {
        if (o) {
            dfs(o.left);
            update(o);
            dfs(o.right);
        }
    }
    function update(o) {
        if (o.val == curnum) {
            count ++;
        } else {
            count = 1;
            curnum = o.val;
        }
        if (count == maxcount) {
            ans.push(curnum);
        } else if (count > maxcount) {
            ans = [curnum];
            maxcount = count;
        }
    }
    dfs(root);
    return ans;
};
[a, b, c ]= [0,0,0];
a = [1,2,3];
d = [...a, 4]
console.log(d)